include/zephyr/drivers/dma.h

```diff
@@ -140,13 +140,15 @@ struct dma_block_config {
  * @typedef dma_callback_t
  * @brief Callback function for DMA transfer completion
  *
- *  If enabled, callback function will be invoked at transfer completion
- *  or when error happens.
- *
- * @param dev Pointer to the DMA device calling the callback.
- * @param user_data A pointer to some user data or NULL
- * @param channel The channel number
- * @param status 0 on success, a negative errno otherwise
+ *  If enabled, callback function will be invoked at transfer completion,
+ *  when water mark is reached or when error happens.
+ *
+ * @param dev           Pointer to the DMA device calling the callback.
+ * @param user_data     A pointer to some user data or NULL
+ * @param channel       The channel number
+ * @param status        - 0-transfer complete
```

Comment:
Please add some defines for the status codes so its easier to read the callback lines.

something like

#define DMA_STATUS_XFER 0
#define DMA_STATUS_BLOCK 1

And use those throughout in place of 0 and 1

Please also note that the STATUS_BLOCK may signify a watermark for when circular buffers are in use in the doc comments here.